### ISSUE TITLE
Harden versioning of dependencies... less problems, more predictable builds

### ISSUE THREAD

**Issue by SilentCicero** [ConsenSys-archive/ganache-cli-archive#200]
I install TestRPC 3.0.0 on one day, and install TestRPC 3.0.0 again on another, and a new issue has popped up. This is due to the versioning accepting newer versions of package dependencies.

Today, I just got a bug from `ethereumjs-tx` but the version of `ethereumjs-testrpc` remains the same. It causes havoc with a test or production system. 

## Expected Behavior
ethereumjs-testrpc dependencies remain the same for every version, so `3.0.0` will always run whether I install it now or 3 months from now.

## Current Behavior
ethereumjs-testrpc dependencies change if a dependency is updated, caused potentially new errors on the same version.

## Possible Solution
Harden the versioning per published build, only upgrade if we explicitly want to upgrade a dependency package version (i.e. remove the '^' character from the versioning).

```json
"dependencies": {
    "async": "1.5.0",
    "bignumber.js": "2.1.4",
    "bip39": "2.2.0",
    "ethereumjs-account": "2.0.4",
    "ethereumjs-block": "1.2.2",
    "ethereumjs-tx": "1.1.0",
    "ethereumjs-util": "4.5.0",
    "ethereumjs-vm": "1.3.0",
    "ethereumjs-wallet": "0.6.0",
    "fake-merkle-patricia-tree": "1.0.1",
    "heap": "0.2.6",
    "merkle-patricia-tree": "2.1.2",
    "seedrandom": "2.4.2",
    "shelljs": "0.6.0",
    "solc": "0.4.2",
    "web3": "0.16.0",
    "web3-provider-engine": "8.1.0",
    "yargs": "3.29.0"
}
```

## Steps to Reproduce (for bugs)
Hard, but essentially install TestRPC right now and see if you get a TypeError from `ethereumjs-tx`.-- https://github.com/ethereumjs/ethereumjs-tx/issues/36

## Context
I'm using testrpc for various projects, the versioning changes are killing me. Some package updates somewhere and my testrpc instance no longer functions upon a fresh re-install of what should be the same package.

## Your Environment
Linux (Ubuntu 16)
node 6+
npm 3+

**Comment by tcoulter** [comment]
Hardening the dependencies this tightly means that I have to release a new version of the TestRPC for every dependency update, even patch versions. I would suggest something tighter but more moderate.

**Comment by johnmiroki** [comment]
@SilentCicero your solution works for me, ie remove the '^' character from the versioning and run `npm install --save`

**Comment by area** [comment]
Replacing `^` with `~` would be a middle ground. The latter upgrades `1.2.x` up to the largest `x` can be, but does not update to `1.3.y`

For what it's worth, we've been caught out by this a lot too.

**Comment by tcoulter** [comment]
@area That's what I was thinking, but forgot the right character at the time (`~` it is). Sounds like a good solution.

**Comment by jmdfm** [comment]
PR: https://github.com/ethereumjs/testrpc/pull/201

**Comment by jmdfm** [comment]
Fixed `ethereumjs-tx` to 1.1.2, and published `testrpc#3.0.1`

**Comment by SilentCicero** [comment]
@tcoulter @johnmcdowall @area it should be very tight per version, use greenkeeper to manage dependency alerts. This is what I am in favour of. Many large production repositories do this.

Anyway, `~` is a better compromise, but I would argue for extreme strictness, due to the critical nature of `testrpc` to the broader testing stack of Ethereum applications.

Just my two cents.
